sdk/eventgrid/azure-eventgrid/azure/eventgrid/_operations/_patch.py

```diff
@@ -187,6 +254,140 @@ def receive_cloud_events(
+            deserialized = response.iter_bytes()
+        else:
+            deserialized = _deserialize(
+                _models._models.PublishResult,  # pylint: disable=protected-access
+                response.json()
+            )
+
+        if cls:
+            return cls(pipeline_response, deserialized, {}) # type: ignore
+
+        return deserialized # type: ignore
+
+
+def _to_http_request(topic_name: str, **kwargs: Any) -> HttpRequest:   
+    # Create a HTTP request for a binary mode CloudEvent
+
+    _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+    _params = case_insensitive_dict(kwargs.pop("params", {}) or {})
+
+    event = kwargs.pop("event")
+
+    # Content of the request is the data, if already in binary - no work needed
+    _content = _check_content_type(event.data)
+    # _content = event.data
+ 
+    # content_type must be CloudEvent DataContentType when in binary mode
+    default_content_type = kwargs.pop('content_type', _headers.pop('content-type', "application/cloudevents+json; charset=utf-8"))
+    content_type: str = event.datacontenttype or default_content_type
+
+    api_version: str = kwargs.pop('api_version', _params.pop('api-version', "2023-06-01-preview"))
+    accept = _headers.pop('Accept', "application/json")
+
+    # Construct URL
+    _url = "/topics/{topicName}:publish"
+    path_format_arguments = {
+        "topicName": _SERIALIZER.url("topic_name", topic_name, 'str'),
+    }
+
+    _url: str = _url.format(**path_format_arguments)  # type: ignore
+
+    # Construct parameters
+    _params['api-version'] = _SERIALIZER.query("api_version", api_version, 'str')
+
+    # Construct headers
+    _headers['content-type'] = _SERIALIZER.header("content_type", content_type, 'str')
+    _headers['Accept'] = _SERIALIZER.header("accept", accept, 'str')
+    # Cloud Headers
+    _headers['ce-source'] = _SERIALIZER.header('ce-source', event.source, 'str')
+    _headers['ce-type'] = _SERIALIZER.header('ce-type', event.type, 'str')
+    if event.specversion:
+        _headers['ce-specversion'] = _SERIALIZER.header('ce-specversion', event.specversion, 'str')
+    if event.id:
+        _headers['ce-id'] = _SERIALIZER.header('ce-id', event.id, 'str')
+    if event.time:
+        _headers['ce-time'] = _SERIALIZER.header('ce-time', event.time, 'str')
+    if event.dataschema:
+        _headers['ce-dataschema'] = _SERIALIZER.header('ce-dataschema', event.dataschema, 'str')
+    if event.subject:
+        _headers['ce-subject'] = _SERIALIZER.header('ce-subject', event.subject, 'str')
+    if event.extensions:
+        for extension, value in event.extensions.items():
+            _headers[f'ce-{extension}'] = _SERIALIZER.header('ce-extensions', value, 'str')
+
+    return HttpRequest(
+        method="POST",
+        url=_url,
+        params=_params,
+        headers=_headers,
+        content=_content, # pass through content
+        **kwargs
+    )
+
+def _check_content_type(data: Any) -> None:
```

Comment:
only accept binary -- let the checking happen at core level and see what happens